components/match2/commons/match_group_input_util.lua

```diff
@@ -1165,4 +1165,49 @@ function MatchGroupInputUtil.standardProcessMatch(match, Parser, mapProps)
 	return match
 end
 
+---@class MapParserInterface
+---@field calculateMapScore? fun(map: table): fun(opponentIndex: integer): integer?
+---@field getExtraData? fun(match: table, games: table, opponents: table[]): table?
+
+--- It may optionally have the following functions:
+--- - calculateMatchScore(map): fun(opponentIndex): integer?
+--- - getExtraData(match, map, opponents): table?
+---@param match table
+---@param opponents table[]
+---@param Parser MapParserInterface
+---@param mapProps any?
+---@return table
+function MatchGroupInputUtil.standardProcessMaps(match, opponents, Parser, mapProps)
+	local maps = {}
+	for key, map in Table.iter.pairsByPrefix(match, 'map', {requireIndex = true}) do
+		local finishedInput = map.finished --[[@as string?]]
+		local winnerInput = map.winner --[[@as string?]]
+
+		map.extradata = Parser.getExtraData and Parser.getExtraData(match, map, opponents) or nil
+		map.finished = MatchGroupInputUtil.mapIsFinished(map)
+
+		local opponentInfo = Array.map(opponents, function(_, opponentIndex)
+			local score, status = MatchGroupInputUtil.computeOpponentScore({
+				walkover = map.walkover,
+				winner = map.winner,
+				opponentIndex = opponentIndex,
+				score = map['score' .. opponentIndex],
+			}, Parser.calculateMapScore and Parser.calculateMapScore(map) or nil)
+			return {score = score, status = status}
+		end)
+
+		map.scores = Array.map(opponentInfo, Operator.property('score'))
+		if map.finished then
+			map.resulttype = MatchGroupInputUtil.getResultType(winnerInput, finishedInput, opponentInfo)
+			map.walkover = MatchGroupInputUtil.getWalkover(map.resulttype, opponentInfo)
+			map.winner = MatchGroupInputUtil.getWinner(map.resulttype, winnerInput, opponentInfo)
```

Comment:
i assume the plan is to switch this to something similar like in match processing?